sql/core/src/test/scala/org/apache/spark/sql/ParametersSuite.scala

```diff
@@ -741,4 +741,21 @@ class ParametersSuite extends QueryTest with SharedSparkSession with PlanTest {
         Row("c1"))
     }
   }
+
+  test("SPARK-50322: parameterized identifier in a sub-query") {
+    withTable("tt1") {
+      sql("create table tt1(c1 int)")
+      sql("insert into tt1 values (1)")
+      def query(p: String): String = {
+        s"""
+          |with v1 as (
+          |  select * from tt1
+          |  where 1 = (Select * from identifier($p))
```

Comment:
```suggestion
          |  where 1 = (select * from identifier($p))
```

Reply:
or we should upper case all SQL keywords

Reply:
I uppercased all SQL keywords in the test.